app/slashcommands/command_custom_status.go

```diff
@@ -83,3 +94,22 @@ func (*CustomStatusProvider) DoCommand(a *app.App, args *model.CommandArgs, mess
 		}),
 	}
 }
+
+func calculateExpiryTime(userTimezone model.StringMap, timezoneEnabled bool) time.Time {
```

Comment:
nit: This method sounds more generic than it is. Could we name it, `calculateEndOfDay` instead?